nibabel/streamlines/trk.py

```diff
@@ -0,0 +1,821 @@
+            properties_size = int(nb_properties * f4_dtype.itemsize)
+
+            # Set the file position at the beginning of the data.
+            f.seek(self.offset_data, os.SEEK_SET)
+
+            # If 'count' field is 0, i.e. not provided, we have to loop
+            # until the EOF.
+            nb_streamlines = self.header[Field.NB_STREAMLINES]
+            if nb_streamlines == 0:
+                nb_streamlines = np.inf
+
+            i = 0
+            nb_pts_dtype = i4_dtype.str[:-1]
+            while i < nb_streamlines:
+                nb_pts_str = f.read(i4_dtype.itemsize)
+
+                # Check if we reached EOF
+                if len(nb_pts_str) == 0:
+                    break
+
+                # Read number of points of the next streamline.
+                nb_pts = struct.unpack(nb_pts_dtype, nb_pts_str)[0]
+
+                # Read streamline's data
+                points_and_scalars = np.ndarray(
+                    shape=(nb_pts, nb_pts_and_scalars),
+                    dtype=f4_dtype,
+                    buffer=f.read(nb_pts * pts_and_scalars_size))
+
+                points = points_and_scalars[:, :3]
+                scalars = points_and_scalars[:, 3:]
+
+                # Read properties
+                properties = np.ndarray(
+                    shape=(self.header[Field.NB_PROPERTIES_PER_STREAMLINE],),
+                    dtype=f4_dtype,
+                    buffer=f.read(properties_size))
+
+                yield points, scalars, properties
+                i += 1
+
+            # In case the 'count' field was not provided.
+            self.header[Field.NB_STREAMLINES] = i
+
+            # Set the file position where it was (in case it was already open).
+            f.seek(start_position, os.SEEK_CUR)
+
+
+class TrkWriter(object):
+    @classmethod
+    def create_empty_header(cls):
+        """ Return an empty compliant TRK header. """
+        header = np.zeros(1, dtype=header_2_dtype)
+
+        # Default values
+        header[Field.MAGIC_NUMBER] = TrkFile.MAGIC_NUMBER
+        header[Field.VOXEL_SIZES] = np.array((1, 1, 1), dtype="f4")
+        header[Field.DIMENSIONS] = np.array((1, 1, 1), dtype="h")
+        header[Field.VOXEL_TO_RASMM] = np.eye(4, dtype="f4")
+        header[Field.VOXEL_ORDER] = b"RAS"
+        header['version'] = 2
+        header['hdr_size'] = TrkFile.HEADER_SIZE
+
+        return header
+
+    def __init__(self, fileobj, header):
+        self.header = self.create_empty_header()
+
+        # Override hdr's fields by those contained in `header`.
+        for k, v in header.items():
+            if k in header_2_dtype.fields.keys():
+                self.header[k] = v
+
+        # By default, the voxel order is LPS.
+        # http://trackvis.org/blog/forum/diffusion-toolkit-usage/interpretation-of-track-point-coordinates
+        if self.header[Field.VOXEL_ORDER] == b"":
+            self.header[Field.VOXEL_ORDER] = b"LPS"
```

Comment:
You say the default order is LPS, but in the `create_empty_header()` function, it is set to RAS... Expected?